bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/library/types/StringListType.java

```diff
@@ -90,15 +87,13 @@ public String toString() {
 
     @Override
     public String toFullString() {
-        List<String> parts = new ArrayList<>(typeDetails.size());
-        for (String row : typeDetails) {
-            parts.add(row.replace(DELIMITER, ESCAPED_DELIMITER));
-        }
-        return StringUtils.join(parts, DELIMITER);
+        return typeDetails.stream()
+            .map(s -> s.replace(DELIMITER, ESCAPED_DELIMITER))
+            .collect(Collectors.joining(",'"));
```

Comment:
looks like a little extra `' ` slipped in here.
Please also use the `DELIMITER` constant.